README.md

```diff
@@ -382,6 +391,21 @@ new event listener whose events are exposed through the Observer handler
 functions and are composable with the various
 [combinators](#operators) available to all Observables.
 
+#### Constructing & converting objects to Observables
+
+Observables can be created by their native constructor, as demonstrated above,
+or by the `Observable.from()` static method. This method constructs a native
+Observable from any of the following objects:
+
+ - `Observable` (in which case it just returns the given object)
+ - `Promise` (or any thenable)
+ - `Iterable` (anything with `Symbol.iterator`)
+ - `AsyncIterable` (anything with `Symbol.asyncIterator`)
+
+Furthermore, any method on the platform that wishes to accept an `Observable`
+can take any of the above objects as well (by accepting a WebIDL `any`), which
+will be converted to a native Observable before being subscribed to.
```

Comment:
This will of course be done in the spec, similarly to what https://streams.spec.whatwg.org/#readable-stream-from-iterable does I imagine.

Also you can imagine passing in an object that satisfies several of the conditions above; e.g., one that's thenable, implements the iterable protocol, _and_ implements the async iterable protocol. In that case, which source which the internal `Obsevable.from()` implementation uses to get the data that it pushes to the created Observable's subscriber is indeed web-observable[^1], so I imagine whatever algorithm that's responsible for doing the conversion must be consistent in which mechanism it prefers (particularly for objects that offer many). That's why in both examples in this PR, I use the same ordering: Observable, Promise, Iterable, AsyncIterable. I think we'd codify this in the spec eventually. Does this at least make sense?

[^1]: That is, whether the `Observable.from()` internals decides to call either `promiseInput.then(next)`, `next(iterableInput[Symbol.iterator]().next())`, or `asyncIterable[Symbol.asyncIterator]().next().then(next)`, if multiple exist.

Reply:
This makes sense. However, I think the order I'd weakly prefer is probably `Observable`, async iterable, iterable, thenable. Reasoning:

- async iterable -> iterable fallback is an established pattern with nice spec support
- Since thenable is a somewhat easier protocol to accidentally fall into, it should be deprioritized in the case of collisions.

Reply:
Done! That sounds good to me. The async iterable -> iterable fallback is news to me, so thanks for chiming in!